File: rpg-encoder/tests/fixtures/csharp/Basic.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MyApp
{
    public class Config
    {
        public string Name { get; set; }
        public string Version { get; set; } = "1.0.0";
        private Dictionary<string, string> _settings = new();

        public void Set(string key, string value)
        {
            _settings[key] = value;
        }

        public string? Get(string key)
        {
            return _settings.TryGetValue(key, out var value) ? value : null;
        }

        public string Process(List<string> data)
        {
            return string.Join("", data);
        }
    }

    public class DataProcessor
    {
        private readonly Config _config;

        public DataProcessor(Config config)
        {
            _config = config;
        }

        public string Run(string input)
        {
            var parts = input.Select(c => c.ToString()).ToList();
            return _config.Process(parts);
        }
    }

    public interface IRepository<T>
    {
        T? Find(long id);
        bool Save(T entity);
        bool Delete(long id);
    }

    public record User(long Id, string Name, string Email);

    public static class Factory
    {
        public static Config CreateConfig(string name)
        {
            return new Config { Name = name };
        }
    }
}

File: rpg-encoder/tests/fixtures/scala/Basic.scala
package com.example.app

import scala.collection.mutable

case class Config(
  name: String,
  version: String = "1.0.0",
  settings: mutable.Map[String, String] = mutable.Map.empty
) {
  def set(key: String, value: String): Config = {
    settings(key) = value
    this
  }

  def get(key: String): Option[String] = settings.get(key)

  def process(data: List[String]): String = data.mkString
}

class DataProcessor(private val config: Config) {
  def run(input: String): String = {
    val parts = input.map(_.toString).toList
    config.process(parts)
  }
}

trait Repository[T] {
  def find(id: Long): Option[T]
  def save(entity: T): Boolean
  def delete(id: Long): Boolean
}

case class User(id: Long, name: String, email: String)

object UserRepository extends Repository[User] {
  private val users = mutable.Map.empty[Long, User]

  override def find(id: Long): Option[User] = users.get(id)
  override def save(entity: User): Boolean = { users(entity.id) = entity; true }
  override def delete(id: Long): Boolean = users.remove(id).isDefined
}

object Factory {
  def createConfig(name: String): Config = Config(name)
}

object Main extends App {
  val config = Factory.createConfig("test")
  println(config.name)
}
